xonsh/proc.py

```diff
@@ -528,3 +528,36 @@ def _wcode_to_popen(code):
     else:
         # Can this happen? Let's find out. Returning None is not an option.
         raise ValueError("Invalid os.wait code: {}".format(code))
+
+
+_CCTuple = namedtuple("_CCTuple", ["stdin",
+                                   "stdout",
+                                   "stderr",
+                                   "pid",
+                                   "returncode",
+                                   "args",
+                                   "alias",
+                                   "stdin_redirect",
+                                   "stdout_redirect",
+                                   "stderr_redirect"])
+
+class CompletedCommand(_CCTuple):
+    """
+    Represents a completed subprocess-mode command.
+    """
+
+    def __bool__(self):
+        return self.returncode == 0
+
+    def __eq__(self, other):
+        return self.returncode == other
+
+    def __hash__(self):
+        return hash(self.returncode)
```

Comment:
Potentially, yes.  But it allows for checking whether the returncode is in a set/dict the same way you can check whether it's in a list/tuple.  Personally, I think the utility outweighs the danger here.